src/lib.rs

```diff
@@ -278,17 +286,23 @@ where
         Key: Borrow<Q>,
         Q: Ord,
     {
-        self.map.contains_key(key) && !self.expired(key)
+        self.peek(key).is_some()
     }
 
     /// Returns the size of the cache, i.e. the number of cached non-expired key-value pairs.
     pub fn len(&self) -> usize {
-        self.map.len() - self.list.iter().take_while(|key| self.expired(key)).count()
+        self.time_to_live.map_or(self.map.len(), |ttl| {
+            let now = Instant::now();
+            self.map.values().filter(|v| v.1 + ttl >= now).count()
+        })
```

Comment:
I think the idea of the original implementation was that counting only the (hopefully few, in practice) expired elements at the beginning of the list is faster than counting all non-expired ones. Not sure whether that's relevant in practice, though.

Reply:
I see.
Again ideally there could be some variable which tells what is supposed to happen (depending on the frequency of get vs number of items).
I can revert

Reply:
I'm not sure about that myself; I just wanted to explain the rationale behind the original code. Your call — or @Fraser999's.